### ISSUE TITLE
add a folding provider

### ISSUE THREAD

**Issue by aeschli** [DotJoshJohnson/vscode-xml#230]
https://github.com/Microsoft/vscode/issues/46200 asks for syntax aware folding for XML.

It would be cool if vscode-xml could implement the foldingRangeProvider API.

**Comment by angelozerr** [comment]
@aeschli  please try https://github.com/angelozerr/lsp4xml/ which implements LSP folding https://github.com/angelozerr/lsp4xml/wiki/Features#folding

**Comment by angelozerr** [comment]
Please try [redhat.vscode-xml](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml) which provides this features:

![image](https://user-images.githubusercontent.com/1932211/46312278-e91c2d80-c5c4-11e8-9b23-cf3ff3a0006c.png)

**Comment by moody31415** [comment]
Are there options that don't require me to install Java? Why does that plugin require
> Java JDK 8 or more recent
> Ensure Java path is set in either: 
> java.home in VSCode preferences
> Environment variable JAVA_HOME or JDK_HOME

**Comment by angelozerr** [comment]
No, you need Java. [redhat.vscode-xml](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml) consumes the [lsp4xml](https://github.com/angelozerr/lsp4xml) language server which is written in Java. The main reason is that, lsp4xml is using [Xerces Java](http://xerces.apache.org/#xerces2-j) which provides a very good `XML Schema support` which is used for completion and validation when XML files are linked to XML Schema.

**Comment by gjsjohnmurray** [comment]
I think this would be a great asset.

**Comment by angelozerr** [comment]
> Are there options that don't require me to install Java? Why does that plugin require

Please note that [vscode-xml 0.15.0 from Red Hat ](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-xml) **no longer require Java**, so you will benefit from folding feature.
